### DISCUSSION TITLE
Error validating: The name of a Field must not start with a number.

### DISCUSSION THREAD

**Discussion by maxganiev** [prisma/prisma#20996] [Q&A]
### Bug description

DB is served by different backend services (via Laravel and Sveltkit via Prisma) so tables can be added both from Laravel and Prisma migrations. I got the table created with Laravel mig. where it has column (Field) names stated as integers.
<img width="553" alt="image" src="https://github.com/prisma/prisma/assets/63904631/d19b3ecc-d657-4209-93c4-4a6ad399e1a0">

When I do `npx prisma db pull` I got that table commented out  and that's OK
<img width="902" alt="image" src="https://github.com/prisma/prisma/assets/63904631/c9b0d1ac-a709-47f3-974e-758a8ea5a93e">

But when later I try to `npx prisma db push` Prisma CLI warns that it will drop all columns from that table. Neither @@ignore nor @map attributes helps:
<img width="794" alt="image" src="https://github.com/prisma/prisma/assets/63904631/45c68ff3-fb73-44a2-b54e-6574823a8246">

How can I either to have this table completely ignored by Prisma or to map column names so it won't drop table columns?

### How to reproduce

1. npx prisma db pull
2. try add ignoring table with @@ignore attribute or by mapping unsupported field names with @map attribute.
3. npx prisma db push

### Expected behavior

Ignored table when pushing or mapped column names

### Prisma information

```prisma
// Add your schema.prisma
```
model oc_product_stocks {
  id          Int     @id @default(autoincrement())
  product_id  Int
  isp         Boolean
  category_id Int
  buhta_id    Int

  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 1 Int @map("1")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 2 Int @map("2")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 3 Int @map("3")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 4 Int @map("4")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 5 Int @map("5")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 6 Int @map("6")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 7 Int @map("7")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 8 Int @map("8")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 9 Int @map("9")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 10 Int @map("10")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 11 Int @map("11")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 12 Int @map("12")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 13 Int @map("13")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 14 Int @map("14")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 16 Int @map("16")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 17 Int @map("17")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 18 Int @map("18")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 19 Int @map("19")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 20 Int @map("20")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 22 Int @map("22")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 23 Int @map("23")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 24 Int @map("24")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // 25 Int @map("25")
  @@ignore
}
```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOs
- Database: MySQL
- Node.js version: 18.12.1

### Prisma Version

```
^4.2.1
```

**Comment by aqrln** [comment]
@maxganiev you need to uncomment the field and provide a valid name for the Prisma Client, as instructed by the comment above. The name of the field in the database will not change as it is specified in the `@map` attribute. For example, if you have
```prisma
/// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
// 1 Int @map("1")
```
you can replace it with something like this:
```prisma
field1 Int @map("1")
```
I understand that the current experience might be confusing, so if you have any suggestions for concrete improvements that would've helped you here, they would be very welcome!

**Reply by ludralph** [reply]
Hi @maxganiev 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by ludralph** [reply]
HI @maxganiev 👋 

Did you get a chance to see @aqrln comment? Do you have any questions about this, In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by maxganiev** [reply]
Hello everyone, I haven't had a chance to test out @aqrln proposal yet, i will put a msg here when I will have given it a try. 
Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
